zummon/src/commonMain/kotlin/companysurvey/Validation.kt

```diff
@@ -202,6 +254,138 @@ class ElectricityValidator : Validator<Electricity> {
+            }
+        } else {
+            ValidationResult(Status.INVALID, translate("electricity.notEnoughValues",
+                electricity.annualElectricityFeedIn_kWh,
+                electricity.quarterHourlyFeedIn_kWh.values.size
+            ))
+        }
+    }
+
+    // quarter-hourly delivery data should not have "holes" longer than 4 days. Display duration of hole in message.
+    fun validateQuarterHourlyDeliveryData(electricity: Electricity): ValidationResult {
+        electricity.quarterHourlyDelivery_kWh?.values ?: return ValidationResult(
+            Status.MISSING_DATA,
+            translate("electricity.quarterHourlyDeliveryDataNotProvided")
+        )
+
+        var maxNullSequence = electricity.quarterHourlyDelivery_kWh.lengthEmptyGapsData()
+
+        return if (maxNullSequence > 384) { // 384 nulls is 4 days of missing data in quarter-hour intervals
+            ValidationResult(
+                Status.INVALID,
+                translate("electricity.quarterHourlyDeliveryDataHolesExceed", maxNullSequence / 4) // Display in hours
+            )
+        } else {
+            ValidationResult(Status.VALID, translate("electricity.quarterHourlyDeliveryDataValid"))
+        }
+    }
+
+    //quarter-hourly production data should not have "holes" longer than 4 days. Display duration of hole in message.
+    fun validateQuarterHourlyProductionData(electricity: Electricity): ValidationResult {
+        electricity.quarterHourlyProduction_kWh?.values ?: return ValidationResult(
+            Status.MISSING_DATA,
+            translate("electricity.quarterHourlyProductionDataNotProvided")
+        )
+
+        var maxNullSequence = electricity.quarterHourlyProduction_kWh.lengthEmptyGapsData()
+
+        return if (maxNullSequence > 384) { // 384 nulls is 4 days of missing data in quarter-hour intervals
+            ValidationResult(
+                Status.INVALID,
+                translate("electricity.quarterHourlyProductionDataHolesExceed", maxNullSequence / 4) // Display in hours
+            )
+        } else {
+            ValidationResult(Status.VALID, translate("electricity.quarterHourlyProductionDataValid"))
+        }
+    }
+
+    //peak of delivery should be less than contracted capacity
+    fun quarterHourlyDeliveryLowContractedCapacity(electricity: Electricity): ValidationResult {
+        val contractedCapacity = (electricity.getContractedConnectionCapacityKw() ?: 0.0).toFloat()
+        val lower = electricity.quarterHourlyDelivery_kWh?.values?.all { (it ?: 0.0).toFloat() <= contractedCapacity } == true
+        return if (lower) {
+            ValidationResult(Status.VALID, translate("electricity.quarterHourlyDeliveryLowContractedCapacityKw", contractedCapacity))
+        } else {
+            ValidationResult(Status.INVALID, translate("electricity.quarterHourlyDeliveryHighContractedCapacityKw", contractedCapacity))
+        }
+    }
```

Comment:
The rule "peak of feed-in should be less than contracted capacity" is checked in the issue but it's not in the PR.

Reply:
is it not resolved with 'every time step in quarter-hourly feed-in should be less than or equal to production + battery power'

Reply:
Understood and done